### DISCUSSION TITLE
同时申请多个证书时，日志会混合在一起

### DISCUSSION THREAD

**Discussion by Tsangkam** [1Panel-dev/1Panel#6482] [Q&A]
### 1Panel 版本

社区版v1.10.6-lts

### 问题描述

同时申请多个证书时，不同证书的日志会混合在一起
![Snipaste_2024-05-02_04-31-18](https://github.com/1Panel-dev/1Panel/assets/40147838/b5c9759c-b384-4a6d-b654-7e04adc16c69)

### 重现步骤

同时申请多个证书

### 期待的正确结果

每个证书申请的日志应当分开显示

**Comment by zhengkunwang223** [comment]
是一个任务中有多个证书，还是创建了多个证书任务？

**Comment by Tsangkam** [comment]
> 是一个任务中有多个证书，还是创建了多个证书任务？

多个证书任务

**Comment by cli-r** [comment]
我也有这个问题，如果同时申请几个证书，日志会被混在一起，并且一旦有申请任务失败，就会一直卡在那里，即使取消任务实际上还是在后台运行，希望能在删除任务时强制关闭正在执行中的申请任务。

**Comment by zhengkunwang223** [comment]
我们这没有复现 是必现吗？

**Comment by yyyytawa** [comment]
我这边也有这个情况 同样证书申请失败删掉重新申请原来日志还在

**Comment by Tsangkam** [comment]
> 我们这没有复现 是必现吗？

我又试了一遍，也是这样的
我同时申请abc三个证书，ab日志都卡在acme注册账号这里，然后就都混合到c申请的日志这里了
![image](https://github.com/1Panel-dev/1Panel/assets/40147838/82f45d4b-df20-4591-a4af-0709d3ba6b5b)
![image](https://github.com/1Panel-dev/1Panel/assets/40147838/7f89803c-ac09-4219-8424-49cbe061a208)

**Comment by zhengkunwang223** [comment]
> > 我们这没有复现 是必现吗？
> 
> 我又试了一遍，也是这样的 我同时申请abc三个证书，ab日志都卡在acme注册账号这里，然后就都混合到c申请的日志这里了 ![image](https://private-user-images.githubusercontent.com/40147838/329756957-82f45d4b-df20-4591-a4af-0709d3ba6b5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxMTI3NzMsIm5iZiI6MTcyMzExMjQ3MywicGF0aCI6Ii80MDE0NzgzOC8zMjk3NTY5NTctODJmNDVkNGItZGYyMC00NTkxLWE0YWYtMDcwOWQzYmE2YjViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDEwMjExM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZjZGVmYjBiNDcyODlmOTRiM2JkN2RkMTZhNWQxMWJhM2MwZGYzMTJmN2YxNWFhMWY1YWNhMzZlNjUxNThmYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hlBKqrF-ASa9qvg7Q3wUZBjRWbfyk62Y0Nw89a_8FbA) ![image](https://private-user-images.githubusercontent.com/40147838/329756961-7f89803c-ac09-4219-8424-49cbe061a208.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxMTI3NzMsIm5iZiI6MTcyMzExMjQ3MywicGF0aCI6Ii80MDE0NzgzOC8zMjk3NTY5NjEtN2Y4OTgwM2MtYWMwOS00MjE5LTg0MjQtNDljYmUwNjFhMjA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDEwMjExM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhYjg0N2QxNGRlMGY1MGNkMTRjNmE4Yjc5OWJhOTQzZWJhNGY4MzRiMTllZjQ2ZGQwMDAyM2UyMjFmYTZjZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.96wNQD3L0xNVyRTBbBHvX6pAamNz-1p2OqQPIGVl1aw)

回访一下 现在还有这个问题吗？我们在近几个版本优化了证书申请

**Reply by icpd** [reply]
v1.10.15-lts
还是有这个这个问题
<img width="1051" alt="image" src="https://github.com/user-attachments/assets/a441c11e-0be5-4906-bc9c-2e1e68304fff">

**Comment by zhengkunwang223** [comment]
看了一下 lego 的代码，lego 不太适合同时申请多个证书，我们会找一下有没有好的解决方案

**Reply by icpd** [reply]
另外我发现 DNS 服务器设置后也是会相互影响，并且没有办法恢复系统默认的 DNS
